vncviewer/DesktopWindow.cxx

```diff
@@ -467,7 +467,32 @@ void DesktopWindow::draw()
     // primary screen rather than the entire window
     if (fullscreen_active() && fullScreenAllMonitors) {
       assert(Fl::screen_count() >= 1);
-      Fl::screen_xywh(sx, sy, sw, sh, 0);
+
+      rfb::Rect viewport_rect, screen_rect;
+      viewport_rect.setXYWH(x(), y(), w(), h());
```

Comment:
Formally the "viewport" is the widget showing the VNC session, and you want the size of the window here. The viewport might be smaller or larger than the window depending on the session.

Reply:
Renamed.